cloud/scope/powervs_machine.go

```diff
@@ -450,9 +461,13 @@ func (m *PowerVSMachineScope) ignitionUserData(userData []byte) ([]byte, error)
 			Ignition: ignV3Types.Ignition{
```

Comment:
lets remove case 2 and in api https://github.com/Karthik-K-N/cluster-api-provider-ibmcloud/blob/e8a4debb187fa62326bce4799d760dbfe7ff1282/api/v1beta2/ibmpowervsmachine_types.go#L147-L148 lets make it default to 3.2 and remove all the older versions since auheader is supported from 3.2 onwards